code/game/objects/items/devices/portable_recharger.dm

```diff
@@ -0,0 +1,337 @@
+		return
+	if(istype(charging, /obj/item/ammo_box/magazine/recharge))
+		var/obj/item/ammo_box/magazine/recharge/power_pack = charging
+		. += span_notice("- \The [charging]'s cell is at <b>[PERCENT(power_pack.stored_ammo.len/power_pack.max_ammo)]%</b>.")
+		return
+	. += span_notice("- \The [charging] is not reporting a power level.")
+
+/obj/item/portable_recharger/Entered(atom/movable/arrived, atom/old_loc, list/atom/old_locs)
+	if(is_type_in_typecache(arrived, allowed_devices))
+		charging = arrived
+		START_PROCESSING(SSmachines, src)
+		finished_recharging = FALSE
+		using_power = TRUE
+		soundloop.start()
+		update_appearance()
+	return ..()
+
+/obj/item/portable_recharger/Exited(atom/movable/gone, direction)
+	if(gone == charging)
+		if(!QDELING(charging))
+			charging.update_appearance()
+		charging = null
+		using_power = FALSE
+		soundloop.stop()
+		update_appearance()
+	if(gone == cell)
+		cell = null
+	return ..()
+
+/obj/item/portable_recharger/attackby(obj/item/attacking_item, mob/user, params)
+	if(hit)
+		return
+
+	if(istype(attacking_item, /obj/item/stock_parts/cell))
+		if(cell)
+			balloon_alert(user, "already loaded!")
+		else
+			if(!user.transferItemToLoc(attacking_item, src))
+				return TRUE
+			cell = attacking_item
+			balloon_alert(user, "installed")
+
+	if(isnull(cell))
+		balloon_alert(user, "no cell!")
+		return ..()
+
+	if(!is_type_in_typecache(attacking_item, allowed_devices))
+		return ..()
+
+	if(charging)
+		return TRUE
+
+	if(!cell.charge)
+		balloon_alert(user, "no charge!")
+		return TRUE
+
+	if(istype(attacking_item, /obj/item/gun/energy))
+		var/obj/item/gun/energy/energy_gun = attacking_item
+		if(!energy_gun.can_charge)
+			balloon_alert(user, "not rechargable!")
+			return TRUE
+	user.transferItemToLoc(attacking_item, src)
+	return TRUE
+
+/obj/item/portable_recharger/screwdriver_act(mob/living/user, obj/item/tool)
+	if(!cell || charging || hit)
+		return FALSE
+
+	if(do_after(user, 3 SECONDS, target = src))
+		cell.update_appearance()
+		cell.forceMove(get_turf(src))
+		balloon_alert(user, "removed [cell]")
+		cell = null
+		tool.play_tool_sound(src, 50)
+		return TRUE
+	return FALSE
+
+/obj/item/portable_recharger/attack_hand(mob/user, list/modifiers)
+	if(loc == user)
+		if(user.get_slot_by_item(src) & slot_flags)
+			take_charging_out(user)
+		else
+			balloon_alert(user, "equip it first!")
+		return TRUE
+
+	add_fingerprint(user)
+	return ..()
+
+///Takes charging item out if there is one
+/obj/item/portable_recharger/proc/take_charging_out(mob/user)
+	if(isnull(charging) || user.put_in_hands(charging))
+		return
+	charging.forceMove(drop_location())
+
+/obj/item/portable_recharger/attack_tk(mob/user)
+	if(isnull(charging))
+		return
+	charging.forceMove(drop_location())
+	return COMPONENT_CANCEL_ATTACK_CHAIN
+
+///Blows up the cell in it
+/obj/item/portable_recharger/proc/detonate()
+	if(!cell)
+		return
+
+	cell.explode()
+
+	charging?.forceMove(drop_location())
+	qdel(src)
```

Comment:
I still recommend implementing `deconstruct()` over doing this

```dm
/obj/item/portable_recharger/deconstruct(dissassembled)
	charging?.forceMove(drop_location())
	return ..()
```
```dm
	cell.explode()
	deconstruct()
```



As it stands if your item is destroyed through other ic means, like poured acid over or something, the gun will be deleted instead of dropping